internal/core/application/market_loader_service_test.go

```diff
@@ -0,0 +1,225 @@
+	"testing"
+)
+
+var (
+	ctx     context.Context
+	filter1 = domain.Filter{
+		Url:        "url1",
+		BaseAsset:  "ba1",
+		QuoteAsset: "qa1",
+	}
+	filter2 = domain.Filter{
+		Url:        "url2",
+		BaseAsset:  "ba2",
+		QuoteAsset: "qa2",
+	}
+	filter3 = domain.Filter{
+		Url:        "url3",
+		BaseAsset:  "ba3",
+		QuoteAsset: "qa3",
+	}
+	page = domain.Page{
+		Number: 1,
+		Size:   20,
+	}
+)
+
+func TestUpdateMarketActiveStatusAndInsertNew(t *testing.T) {
+	type fields struct {
+		marketRepository domain.MarketRepository
+	}
+	type args struct {
+		activeMarkets []domain.Market
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		wantErr     bool
+		prepareData func(repo domain.MarketRepository) ([]domain.Market, error)
+		validate    func(t *testing.T, repo domain.MarketRepository)
+	}{
+		{
+			name: "test_1",
+			fields: fields{
+				marketRepository: inmemory.NewRepository(),
+			},
+			args: args{
+				activeMarkets: []domain.Market{
+					{
+						ProviderName: "pn2",
+						Url:          "url2",
+						BaseAsset:    "ba2",
+						QuoteAsset:   "qa2",
+						Active:       true,
+					},
+				},
+			},
+			wantErr: false,
+			prepareData: func(repo domain.MarketRepository) ([]domain.Market, error) {
+				for _, v := range prepareMarkets1() {
+					if err := repo.InsertMarket(ctx, v); err != nil {
+						return nil, err
+					}
+				}
+
+				return prepareMarkets1(), nil
+			},
+			validate: func(t *testing.T, repo domain.MarketRepository) {
+				m1, err := repo.GetAllMarketsForFilter(ctx, []domain.Filter{filter1}, page)
+				require.NoError(t, err)
+				require.Equal(t, false, m1[0].Active)
+
+				m2, err := repo.GetAllMarketsForFilter(ctx, []domain.Filter{filter2}, page)
+				require.NoError(t, err)
+				require.Equal(t, true, m2[0].Active)
+
+				m3, err := repo.GetAllMarketsForFilter(ctx, []domain.Filter{filter3}, page)
+				require.NoError(t, err)
+				require.Equal(t, false, m3[0].Active)
+			},
+		},
+		{
+			name: "test_2",
+			fields: fields{
+				marketRepository: inmemory.NewRepository(),
+			},
+			args: args{
+				activeMarkets: []domain.Market{},
+			},
+			wantErr: false,
+			prepareData: func(repo domain.MarketRepository) ([]domain.Market, error) {
+				for _, v := range prepareMarkets1() {
+					if err := repo.InsertMarket(ctx, v); err != nil {
+						return nil, err
+					}
+				}
+
+				return prepareMarkets1(), nil
+			},
+			validate: func(t *testing.T, repo domain.MarketRepository) {
+				m1, err := repo.GetAllMarketsForFilter(ctx, []domain.Filter{filter1}, page)
+				require.NoError(t, err)
+				require.Equal(t, false, m1[0].Active)
+
+				m2, err := repo.GetAllMarketsForFilter(ctx, []domain.Filter{filter2}, page)
+				require.NoError(t, err)
+				require.Equal(t, false, m2[0].Active)
+
+				m3, err := repo.GetAllMarketsForFilter(ctx, []domain.Filter{filter3}, page)
+				require.NoError(t, err)
+				require.Equal(t, false, m3[0].Active)
+			},
+		},
+		{
+			name: "test_3",
```

Comment:
This seems to test only that prev inactive markets can be set to active. Are we missing testing that new markets are added without problems?
For this purpose, we should just add a brand new market in `activeMarkets` list